io/io/src/TMemFile.cxx

```diff
@@ -404,7 +402,7 @@ Int_t TMemFile::SysReadImpl(Int_t, void *buf, Long64_t len)
 {
    TRACE("READ")
 
-   if (fBlockList.fBuffer == 0) {
+   if (fBlockSeek == nullptr || fBlockSeek->fBuffer == nullptr) {
```

Comment:
Why is this change needed?

Reply:
It's a drive-by improvement: `fBlockSeek` and `fBlockSeek->fBuffer` are dereferenced below, so they should be checked (not `fBlockList.fBuffer` which is "contractually connected" to `fBlockSeek->fBuffer`).